### DISCUSSION TITLE
Is mruby sefl contained?

### DISCUSSION THREAD

**Discussion by voxik** [mruby/mruby#6242] [Q&A]
As far as I understand, mruby needs Ruby for build, because the build is done using Rake. But is there a (theoretical) way to build mruby without Ruby?

I am asking, because the MRI is likely impossible to build without Ruby these days, so I wonder if mruby is self contained and once having mruby, could it be used to bootstrap MRI?

**Comment by matz** [comment]
I am sorry but mruby requires Ruby to be compiled. I have no concrete plan to address it because existing build tools are too weak and programmable.
